### PR TITLE
Increase test cases & coverage

### PR DESCRIPTION
**Pull request by jGaboardi**
This PR begins to address #112.

* blackens repo
* Clean up (some) docstrings
* Streamlines tests by reusing objects
* Fills out testing suite
  * [x] `azp`
    * basic AZP
    * ~~simulated annealing AZP~~ **code needs attention**
    * ~~Tabu AZP~~ **code needs attention**
      * ~~basic tabu~~ **code needs attention**
      * ~~reactive tabu~~  **code needs attention**
  * [x] `maxp`
    * `verbose == True`
    * complex cases (e.g., check move and simulated annealing)
    * one variable column for `attrs_name`
  * [x] `random_regions`
    * random regions
    * random region 
  * [x] `region_k_means`
    * reformatting and cleaning of test
    * add complex lattice test
  * [x] `skater`
    * test verbose
    * fix and test super verbose
    * add non-default parameters test
    * test islands (pass & fail)
  * [x] `spenc`
    * defaults, gamma, n_clusters
    * `spenc_lib`
  * [x] `ward` already 100% covered
    * reformatting and cleaning of test
  * [x] `util`
    * test solver checks

### CODE REVIEW

**Comment by jGaboardi** [comment]
I think this is about as good as we are going to get at the moment. The vast majority of code that is actually being used in regionalization models is now covered in test. The bigger issue is that there is a large amount of code that is simply not being used.

File: spopt/region/azp_util.py [hunk 1]
```diff
@@ -152,35 +147,31 @@ class AllowMoveAZPMaxPRegions(AllowMoveStrategy):
...
         """
 
         Parameters
         ----------
-        spatially_extensive_attr : :class:`numpy.ndarray`, default: None
+        spatially_extensive_attr : numpy.ndarray
             See corresponding argument in
-            :meth:`region.max_p_regions.heuristics.MaxPRegionsHeu.fit_from_scipy_sparse_matrix`.
-        threshold : numbers.Real or :class:`numpy.ndarray`
+            ``region.max_p_regions.heuristics.MaxPRegionsHeu.fit_from_scipy_sparse_matrix``.
```
Comment (knaaptime) [inline spopt/region/azp_util.py:157]: just a note to self, really, but we probably want a better description here that doesnt need to references region

Comment (jGaboardi) [reply spopt/region/azp_util.py:157]: @knaaptime completely agree. When I started going through everything, I realized the docs will need a lot of love, so I focused on the testing coverage for his PR.

File: spopt/region/region_k_means.py [hunk 2]
```diff
@@ -133,14 +133,14 @@ class RegionKMeansHeuristic(BaseSpOptHeuristicSolver):
 
     Attributes
     ----------
-    
-    labels_ : 
+
```
Comment (knaaptime) [inline spopt/region/region_k_means.py:136]: another note to self we'll wanna fill these in at some point

File: spopt/region/skater.py [hunk 3]
```diff
@@ -318,10 +305,10 @@ def tqdm(noop, desc=""):
 
     def make_cut(self, in_node, out_node, score, MSF=None):
         """
-        make a cut on the MSF inplace, provided the in_node, out_node, MSF, and score. 
+        make a cut on the MSF inplace, provided the in_node, out_node, MSF, and score.
```
Comment (knaaptime) [inline spopt/region/skater.py:308]: nbd, but we're switching docstring format in this function?

File: spopt/region/azp_util.py (hunk 1 above)
Comment (knaaptime) [reply spopt/region/azp_util.py:157]: lets do it in phases

**Pull request merged**